configure.ac

```diff
@@ -144,12 +144,32 @@ AS_IF([test "x$enable_coretext" != xno], [
 AS_IF([test "x$enable_directwrite" != xno], [
     # Linking to DirectWrite directly only works from Windows
     AC_MSG_CHECKING([for DIRECTWRITE])
-    AC_LINK_IFELSE([
+    AC_PREPROC_IFELSE([
         AC_LANG_PROGRAM([[#include <windows.h>]], [[;]])
     ], [
-        AC_DEFINE(CONFIG_DIRECTWRITE, 1, [found DirectWrite])
         directwrite=true
         AC_MSG_RESULT([yes])
+        AC_MSG_CHECKING([for Win32 desktop APIs])
+        AC_PREPROC_IFELSE([
+            AC_LANG_PROGRAM([[
+                #include <winapifamily.h>
+                #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+                #error Win32 desktop APIs are available
+                #endif
+            ]], [[;]])
```

Comment:
Interesting observation. I can reorder the checks, but the `./configure` output would become less clear. On the bright side, it would save one check on some Windows setups. Oh, but on the other hand, it would add an extra check on all non-Windows platforms, which is worse.

Note there’s only two checks in total: the new check can’t replace the old one entirely but works in tandem with it to refine the detection. We need to distinguish between three cases here: desktop Windows, app Windows, and non-Windows. A single check can’t do that, I don’t think.

Reply:
I guess it's better to keep the `windows.h` check then. In that case just a note: if you decide to use a compile test afterall there's the option to just use `AC_CHECK_HEADER` for a more compact `configure.ac` — but instead of the current custom check-message it would be a standard header-check message.